src/main/java/com/skyevel/universalscoreboard/event/player/AsyncPlayerBoardUpdateEvent.java

```diff
@@ -1,75 +1,93 @@
 import org.bukkit.event.HandlerList;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.ArrayList;
 import java.util.List;
 
+/**
+ * Event that is called when a player's scoreboard is updated asynchronously.
+ * <p>
+ *   Now we don't recommend using Strings for updating lines, when using Strings
+ *   the input will go through a processor to serialize it to a {@link Component}.
+ *   For more information: <a href="https://docs.advntr.dev/">Kyori's Adventure API</a>
+ *   <br><br>
+ *   <i>To get help with {@link Component}s, you can use the {@link ComponentUtil} class.</i>
+ * </p>
+ * @see ComponentUtil
+ * @see FastBoard
+ */
 public class AsyncPlayerBoardUpdateEvent extends Event {
+  private static final HandlerList HANDLERS = new HandlerList();
+  private final FastBoard board; // the FastBoard instance
+  private List<Component> lines; // the initial lines
 
-    private static final HandlerList HANDLERS = new HandlerList();
-    private List<String> lines = new ArrayList<>();
-    private final FastBoard board;
+  public AsyncPlayerBoardUpdateEvent(FastBoard board) {
+    this.board = board;
+    this.lines = board.getLines();
+  }
 
-    /**
-     * Get the Handler list of the Event.
-     *
-     * @return a list of the handlers
-     */
-    @Override
-    public @NotNull HandlerList getHandlers() {
-        return HANDLERS;
-    }
+  public static HandlerList getHandlerList() {
+    return HANDLERS;
+  }
 
-    public static HandlerList getHandlerList() {
-        return HANDLERS;
-    }
+  /**
+   * Get the Handler list of the Event.
+   *
+   * @return a list of the handlers
+   */
+  @Override
+  public @NotNull HandlerList getHandlers() {
+    return HANDLERS;
+  }
 
-    public AsyncPlayerBoardUpdateEvent(FastBoard board) {
-        this.board = board;
-    }
+  /**
+   * Get the Player who is attached to the current Scoreboard.
+   *
+   * @return the player
+   */
+  public Player getPlayer() {
+    return this.board.getPlayer();
+  }
 
-    /**
-     * Get the Player who is attached to the current Scoreboard.
-     *
-     * @return the player
-     */
-    public Player getPlayer() {
-        return this.board.getPlayer();
-    }
+  /**
+   * Get the lines of the current Scoreboard.
+   *
+   * @return a list of {@link Component}s
+   */
+  public List<Component> getLinesAsComponents() {
+    return this.lines;
+  }
 
-    /**
-     * Get the lines of the current Scoreboard.
-     *
-     * @return a list of Strings
-     */
-    public List<String> getLines() {
-        return lines;
-    }
-
-    /**
-     * Overwrite the lines of the current Scoreboard.
-     *
-     * @param lines a list of Strings
-     */
-    public void setLines(List<String> lines) {
-
-        this.lines = lines;
-    }
-
-    /**
-     * Overwrite the lines of the current Scoreboard.
-     *
-     * @param lines a listing of Strings
-     */
-    public void setLines(String... lines) {
-        this.lines = List.of(lines);
-    }
```

Comment:
Re-add setLines(String... lines) and also setLines(List<String>) with @Deprecated annotation attached to both of them to maintain backward compatibility.

Reply:
Yeah so I didn't intend to delete the setLines(List) for String nor Component and deprecation is a musn't thing because there is a chance that a developer would use it in their logic.

Reply:
Resolved in [d25da15](https://github.com/SkyEvel/UniversalScoreboard/pull/1/commits/d25da15e3f224e4e2df5fd9abfda8dec303e4e45)